### PR TITLE
Set timeout of the provisioned appliance and ensure we have browser_steal

### PR DESCRIPTION
**Pull request by mfalesni**
{{pytest: cfme/tests/control/test_compliance.py -v --long-running --use-provider vsphere55 --use-provider rhevm33 }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 1c8efe301fead954a25ef3036aa4906e4c9a905d:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/tests/control/test_compliance.py [hunk 1]
```diff
@@ -62,7 +62,9 @@ def compliance_vm(request, provider_key, provider_crud):
         appliance.configure(setup_fleece=True)
         vm = Vm(appliance.vm_name, provider_crud)
     # Do the final touches
-    with appliance.ipapp:
+    appliance.ipapp.browser_steal = True
+    with appliance.ipapp as appl:
+        appl.set_session_timeout(86400)
```
Comment (seandst) [inline cfme/tests/control/test_compliance.py]: It occurs to me that we should probably be doing this as a result of opening the browser, logging in, or some other appropriate event. I also think it's likely that using the `IPAppliance` context manager interface should be all you need, and we should automatically set `browser_steal` when using it, or accept `browser_steal` as an argument to the context manager.

Comment (mfalesni) [inline cfme/tests/control/test_compliance.py]: IPAppliance does not automatically set browser_steal. Adding the argument would btw mean adding a __call__ method that would accept that parameter.

File: utils/appliance.py [hunk 2]
```diff
@@ -1364,6 +1364,18 @@ def get_yaml_config(self, config_name):
     def set_yaml_config(self, config_name, data_dict):
         return db.set_yaml_config(config_name, data_dict, self.address)
 
+    def set_session_timeout(self, timeout=86400):
+        try:
+            vmdb_config = self.get_yaml_config("vmdb")
+            if vmdb_config["session"]["timeout"] < timeout:
```
Comment (seandst) [inline utils/appliance.py]: `<` made sense in the original implementation, when we had a high degree of certainty that we were acting on a new appliance. Adding this as a method to `IPAppliance` opens it up to be used at any time, and should therefore use `!=`.

File: utils/appliance.py [hunk 3]
```diff
@@ -1364,6 +1364,18 @@ def get_yaml_config(self, config_name):
...
+                vmdb_config["session"]["timeout"] = timeout
+                self.set_yaml_config("vmdb", vmdb_config)
+        except Exception as ex:
+            # Definitely need to implement retries or something, this is
+            # just a bandaid
+            logger.error('Setting session timout failed')
+            logger.exception(ex)
```
Comment (seandst) [inline utils/appliance.py]: This little hack I put in was tough to justify as a fixture, but in this method it's even less good...

Unfortunately, I think it's still necessary, so a good compromise would at least be to allow the caller to specify if this should raise an exception or only log it, with the default probably being the current behavior (logging it), and other usages inside tests raising exceptions.

**Comment by dajoRH** [comment]
Lint report for commit da9679691e4d043ef9d92778345d8eee215c645e:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by seandst** [comment]
:+1: punch cakes

**Comment by dajoRH** [comment]
Lint report for commit e6fcfc225c9b5465568c5700030ea2af07ba05ad:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by mfalesni** [comment]
@seandst Will you merge? Seems it does not solve the problems but at least it does not break anything else.

**Pull request merged**